java/core/page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/support/capabilities/SelenoidCapabilitiesParser.java

```diff
@@ -0,0 +1,95 @@
+
+public class SelenoidCapabilitiesParser implements CapabilitiesParser {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SelenoidCapabilitiesParser.class);
+
+    private static final Configuration PROPERTIES = Properties.getProperties();
+    private static final String VIDEONAME_FORMAT = UUID.randomUUID().toString() + "-%s";
+
+    private final DesiredCapabilities capabilities = new DesiredCapabilities();
+
+    @Override
+    public DesiredCapabilities parse() {
+        setVersion(PROPERTIES.getSelenoidBrowserVersion());
+        
+        setCapability("enableVNC", PROPERTIES.getSelenoidEnableVNC());
+        setCapability("screenResolution", PROPERTIES.getSelenoidScreenResolution());
+        setCapability("enableVideo", PROPERTIES.getSelenoidEnableVideo());
+        setCapability("videoName", PROPERTIES.getSelenoidVideoName(), VIDEONAME_FORMAT);
+        setCapability("videoScreenSize", PROPERTIES.getSelenoidVideoScreenSize());
+        setCapability("videoFrameRate", PROPERTIES.getSelenoidVideoFrameRate());
+        setCapability("name", PROPERTIES.getSelenoidNameOfTests());
+        setCapability("timeZone", PROPERTIES.getSelenoidTimeZone());
+        setCapability("hostsEntries", PROPERTIES.getSelenoidHostEntries());
+        setCapability("applicationContainers", PROPERTIES.getSelenoidApplicationContainers());
+        setCapability("labels", PROPERTIES.getSelenoidContainerLables());
+
+        setBrowserDefaultCapabilities();
+
+        return capabilities;
+    }
+
+    private void setCapability(String capabilityName, String capabilityValue, String format) {
+        if (!capabilityValue.isEmpty()) {
+            capabilities.setCapability(capabilityName, String.format(format, capabilityValue));
+        } else {
+            LOG.info("Capability \"{}\" for Selenoid isn't set. Using default capability.", capabilityName);
+        }
+    }
+
+    private void setCapability(String capabilityName, String capabilityValue) {
+        setCapability(capabilityName, capabilityValue, "%s");
+    }
+
+    private void setCapability(String capabilityName, boolean capabilityValue) {
+        setCapability(capabilityName, String.valueOf(capabilityValue), "%s");
+    }
+
+    private void setVersion(String capabilityValue) {
+        if (!capabilityValue.isEmpty()) {
+            capabilities.setVersion(capabilityValue);
+        } else {
+            LOG.info("Capability \"browserVersion\" for Selenoid isn't set. Using default capability.");
+        }
+    }
+
+    private void setBrowserDefaultCapabilities() {
+        switch (TagWebDriver.getBrowserName().toLowerCase()) {
+            case OPERA:
+                handleOpera();
+                break;
+            case IE:
+                handleIe();
+                break;
+            default:
+                break;
+        }
+    }
+
+    private void handleOpera() {
+        capabilities.setCapability("operaOptions", new HashMap<String, String>() {
+            {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Move the contents of this initializer to a standard constructor or to field initializers. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1171)